packages/appium/lib/schema/schema.js

```diff
@@ -4,11 +4,40 @@ import Ajv from 'ajv';
 import addFormats from 'ajv-formats';
 import _ from 'lodash';
 import path from 'path';
-import { DRIVER_TYPE, PLUGIN_TYPE } from '../extension-config';
-import { ReadonlyMap } from '../utils';
+import {DRIVER_TYPE, PLUGIN_TYPE} from '../constants';
 import appiumConfigSchema from './appium-config-schema';
-import { APPIUM_CONFIG_SCHEMA_ID, ArgSpec, SERVER_PROP_NAME } from './arg-spec';
-import { keywords } from './keywords';
+import {APPIUM_CONFIG_SCHEMA_ID, ArgSpec, SERVER_PROP_NAME} from './arg-spec';
+import {keywords} from './keywords';
+
+/**
+ * A Map where you can set properties, but only once. And you can't remove anything. So there.
+ * @template K,V
+ * @extends {Map<K,V>}
+ */
+export class ReadonlyMap extends Map {
+  /**
+   * @param {K} key
+   * @param {V} value
+   */
+  set (key, value) {
```

Comment:
IMHO it's confusing to have `set` in an immutable map. I'd rather prefer an implementation where a map can only be initialised upon creation like in https://immutable-js.com/#javaScript-first-api

Reply:
I think this is a misnomer. I probably need to rename it.  It's _not_ an immutable Map.  It's a readable map which only supports "inserts"; no delete and no update.    I don't really know what to call that.

Reply:
I've renamed it 😄